### PR TITLE
Optimize multiplyBasisFunction by caching math operations

### PR DESCRIPTION
**Pull request by Connormiha**
I have optimized some operations to avoid re-computation.

### CODE REVIEW

File: TypeScript/src/encode.ts [hunk 1]
```diff
@@ -18,14 +18,17 @@ const multiplyBasisFunction = (
   const bytesPerRow = width * bytesPerPixel;
 
   for (let x = 0; x < width; x++) {
+    const bytesPerPixelX = bytesPerPixel * x;
+
     for (let y = 0; y < height; y++) {
+      const basePixelIndex = bytesPerPixelX + (y * bytesPerRow);
```
Comment (Thisen) [inline TypeScript/src/encode.ts:24]: Can you remove the parenteses?

**Review by Thisen — CHANGES_REQUESTED** [review]
A small change, then LGTM :)

File: TypeScript/src/encode.ts (hunk 1 above)
Comment (Connormiha) [reply TypeScript/src/encode.ts]: Done

**Review by Thisen — APPROVED** [review]
Thanks for you contribution :)

**Pull request merged**